src/figuresClasses.ts

```diff
@@ -1,11 +1,92 @@
-export interface Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public firstSide: number,
+    public secondSide: number,
+    public thirdSide: number,
+  ) {
+    const biggestSide: number = Math.max(firstSide, secondSide, thirdSide);
+    const otherSidesSum: number = [firstSide, secondSide, thirdSide]
+      .filter((side) => side !== biggestSide)
+      .reduce((acc, side) => acc + side);
+
+    if (firstSide <= 0 || secondSide <= 0 || thirdSide <= 0) {
+      throw new Error('all sides must be positive');
+    }
+
+    if (biggestSide >= otherSidesSum) {
```

Comment:
[CODE STYLE] - Similar to the previous comment, the error message should be more detailed. A better message would be 'The provided sides do not form a valid triangle', which explains the geometric rule that the sum of the lengths of any two sides must be greater than the length of the remaining side.